### PR TITLE
feat: enable batching for matches2docrankers

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/drivers/rank/__init__.py [hunk 1]
```diff
@@ -86,18 +88,26 @@ def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
...
-            if not old_match_scores:
-                continue
+            old_scores.append(old_match_scores)
+            queries_metas.append(query_meta)
+            matches_metas.append(match_meta)
 
-            new_scores = self.exec_fn(old_match_scores, query_meta, match_meta)
-            if num_matches != len(new_scores):
+        new_scores = self.exec_fn(old_scores, queries_metas, matches_metas)
+        if len(new_scores) != len(new_scores):
```
Comment (nan-wang) [inline jina/drivers/rank/__init__.py:96]:
```suggestion
        if len(new_scores) != len(docs):
```

File: jina/resources/executors.requests.Match2DocRanker.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+on:
+  ControlRequest:
+    - !ControlReqDriver {}
+  SearchRequest:
+    with:
+      traversal_paths: [ 'r' ]
```
Comment (nan-wang) [inline jina/resources/executors.requests.Match2DocRanker.yml:6]: Do we need this? This is the same as the default value.

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: tests/integration/evaluation/rank/yaml/rank.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ metas:
   py_modules: dummy_ranker.py
 requests:
   on:
-    [ SearchRequest ]:
+    [SearchRequest]:
```
Comment (nan-wang) [inline tests/integration/evaluation/rank/yaml/rank.yml:6]:
```suggestion
    SearchRequest:
```

File: jina/resources/executors.requests.Match2DocRanker.yml (hunk 2 above)
Comment (JoanFM) [reply jina/resources/executors.requests.Match2DocRanker.yml]: Done!

File: jina/drivers/rank/__init__.py (hunk 1 above)
Comment (JoanFM) [inline jina/drivers/rank/__init__.py:96]:
```suggestion
        if len(new_scores) != len(docs):
```

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**